internal/install/static_kibana_healthcheck_sh.go

```diff
@@ -0,0 +1,11 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package install
+
+const kibanaHealthcheckSh = `#!/bin/bash
+
+curl -s -f http://127.0.0.1:5601/login | grep kbn-injected-metadata 2>&1 >/dev/null
+curl -s -f -u elastic:changeme http://elasticsearch:9200/_cat/shards | grep .security | grep STARTED
```

Comment:
This checks if _any_ shard is `STARTED`. That could be potentially misleading as a replica might've started but not a primary (I know, not possible on a 1-node cluster but still). It could also be misleading in case the security index ever decides to have multiple primary shards, not just one, and some of them have not yet started.

A more robust test would be to check if _all_ primary shards have been `STARTED`, which is another way of saying that the index is not red (i.e. it is either yellow or green).

```suggestion
curl -s -f -u elastic:changeme "http://elasticsearch:9200/_cat/indices/.security-*?h=health" | grep -v red
```

Reply:
Thanks!